modules/rostests/apitests/isapnp/tests.c

```diff
@@ -0,0 +1,369 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Unit Tests for the ISA PnP bus driver (device discovery and resource tests)
+ * COPYRIGHT:   Copyright 2024 Dmitry Borisov <di.sean@protonmail.com>
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include "precomp.h"
+
+#include "../../../../drivers/bus/isapnp/isapnp.c"
+#include "../../../../drivers/bus/isapnp/hardware.c"
```

Comment:
You can add these to CMakeLists too.

Reply:
Well, one common way is to include the .c file in the unit test code.